server/chat-plugins/battlesearch.ts

```diff
@@ -0,0 +1,412 @@
+/**
+ * Battle search - handles searching battle logs.
+ */
+import * as child_process from 'child_process';
+import * as util from 'util';
+
+import {FS} from '../../lib/fs';
+import {Utils} from '../../lib/utils';
+import {QueryProcessManager} from '../../lib/process-manager';
+import {Repl} from '../../lib/repl';
+import {Dex} from '../../sim/dex';
+import {Config} from '../config-loader';
+import {checkRipgrepAvailability} from '../modlog';
```

Comment:
I guess this is why they were the same file in the past.

I wonder if there's some global we can stick this in. Like `Chat` or something.

Reply:
tbh we could probably just use the existing `Config` setting that it generates instead of the function, no?

Reply:
Oh yeah, `config-loader` already exists; we can probably stick it there?

Reply:
Done.